### ISSUE TITLE
Implement Hash, PartialOrd, Eq, Ord for types

### ISSUE THREAD

**Issue by PonasKovas** [CrabCraftDev/CrabNBT#22]
Hello, it would be awesome if the Nbt types could implement these traits, if possible.

And also, quick unrelated question - can the nbt writing methods ever fail with any error other than IO? maybe they should return `Result<..., io::Error>` instead of `crab_nbt::Error`?

**Comment by goteusz-maszyk** [comment]
They can return other errors. See https://github.com/CrabCraftDev/CrabNBT/blob/main/src/error.rs

**Comment by goteusz-maszyk** [comment]
Well, hashmaps niether do support hashing nor ordering. We cannot also implement Eq because of [floats](https://www.reddit.com/r/rust/comments/2h86qk/dealing_with_floats_and_lack_of_eq/).

**Comment by SzczurekYT** [comment]
Poor research, not all of them, but some of the traits can be actually implemented.

**Comment by SzczurekYT** [comment]
TLDR: `PartialOrd` has been implemented and other traits can't be implemented because of floats
